libvast/src/system/accountant.cpp

```diff
@@ -230,18 +235,16 @@ struct accountant_state_impl {
     if (auto err = dest.send(
           std::span<char>{reinterpret_cast<char*>(buf.data()), buf.size()},
           timeout_usec)) {
-      switch (ec{err.code()}) {
-        case ec::timeout: {
-          uds_datagram_sink_dropping = true;
-          break;
-        }
-        default: {
-          VAST_WARN("{} failed to write metrics to UDS sink: {}", *self, err);
-          VAST_WARN("{} turns off metrics reporting via UDS", *self);
-          uds_datagram_sink.reset();
-          return;
-        }
+      const auto code = ec{err.code()};
+      if (code == ec::timeout)
```

Comment:
You should be able to just say `if (err == ec::timeout)`, regardless of whether or not the error has a context in addition to the code.

Reply:
`ec{err.code()}` is needed a bit further down anyways so why not give it a name and avoid the indirection through an extra `operator!=`? I don't think it makes a difference in readability.

Reply:
Ah, didn't see that use of it. Feel free to resolve.